mule-extensions-api/src/main/java/org/mule/runtime/extension/api/values/SdkConfigurationParameterValueProvider.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.values;
+
+import org.mule.api.annotation.NoImplement;
+import org.mule.sdk.api.values.Value;
+import org.mule.sdk.api.values.ValueResolvingException;
+
+import java.util.Set;
+
+/**
+ * This interface allows the exposure of the {@link Set} of {@link Value values} associated to a Configuration's or Connection
+ * Provider's parameter.
+ *
+ * @since 1.4
+ */
+@NoImplement
+public interface SdkConfigurationParameterValueProvider {
```

Comment:
shouldn't this class be actually defined in `mule-sdk-api`? Also, if this is introduced, the existing `ConfigurationParameterValueProvider` should be deprecated pointing to the new interface

Reply:
Yes, `ConfigurationParameterValueProvider` should be deprecated, right now it only has some methods marked as deprecated.

This new interface `SdkConfigurationParameterValueProvider` won't be used by the connector developers.

Reply:
who uses this interface? It doens't look like this deprecation/replacement is completed.  What's the new API to be used from now on and where's the legacy API which keeps backwards compatibility?

Reply:
It replaces `ConfigurationParameterValueProvider`, which uses the legacy Value interface. It's used by `MuleValueProviderService`, I haven't found any other place where it's used.

`SdkConfigurationParameterValueProvider` uses the new Value interface and removes the deprecated methods.

Reply:
This is API. Cannot be removed or broken, even if your search comes back empty. You don't know if any other product is using it and would make it hard for other teams to write code that works in all versions.

Also, the SDK Value class is only to be used in connectors. External clients should continue to use the one in mule-api.  I also see no reason why to create this class instead of simpy adding new methods in the one we have and deprecating the old methods